launcher/ui/dialogs/ProfileSetupDialog.cpp

```diff
@@ -261,7 +264,21 @@ void ProfileSetupDialog::setupProfileFinished()
     } else {
         auto parsedError = MojangError::fromJSON(*m_profile_response);
         ui->errorLabel->setVisible(true);
-        ui->errorLabel->setText(tr("The server returned the following error:") + "\n\n" + parsedError.errorMessage);
+
+        QString errorMessage = tr("Network error(%1): %2 ;HTTP Status: %3")
+                                   .arg(QString::number(m_profile_task->error()), m_profile_task->errorString(),
+                                        QString::number(m_profile_task->replyStatusCode()));
+
+        if (parsedError.fullyParsed) {
+            errorMessage += "Path: " + parsedError.path + "\n";
+            errorMessage += "Error: " + parsedError.error + "\n";
+            errorMessage += "Error message: " + parsedError.errorMessage + "\n";
```

Comment:
```suggestion
            errorMessage += "Message: " + parsedError.errorMessage + "\n";
```
Could also call it "Response"